### PR TITLE
[SW-256] Add send process indication

### PR DESCRIPTION
**Pull request by peronczyk**

### CODE REVIEW

File: src/popup/router/components/TransferReview.vue [hunk 1]
```diff
@@ -141,102 +141,97 @@ export default {
...
-        this.$store.dispatch('modals/open', {
-          name: MODAL_DEFAULT,
-          title: this.$t('modals.transaction-failed.msg'),
-          icon: 'critical',
-        });
+        this.openTransactionFailedModal();
         throw e;
       } finally {
         this.loading = false;
+        this.$emit('success');
```
Comment (pstachel) [inline src/popup/router/components/TransferReview.vue:232]: should we emit "success" even if the error was thrown?

File: src/popup/router/components/TransferReview.vue [hunk 2]
```diff
@@ -279,18 +274,20 @@ export default {
...
-          name: 'default',
-          title: this.$t('modals.transaction-failed.msg'),
-          icon: 'critical',
-        });
+        this.openTransactionFailedModal();
         e.payload = { url: recipient };
         throw e;
       } finally {
         this.loading = false;
+        this.$emit('success');
```
Comment (pstachel) [inline src/popup/router/components/TransferReview.vue:282]: should we emit "success" even if the error was thrown?

**Review by pstachel — CHANGES_REQUESTED** [review]

File: src/popup/router/components/TransferReview.vue (hunk 1 above)
Comment (peronczyk) [reply src/popup/router/components/TransferReview.vue]: You have right, it should not. Fixed...

File: src/popup/router/components/TransferReview.vue (hunk 2 above)
Comment (peronczyk) [reply src/popup/router/components/TransferReview.vue]: You have right, it should not. Fixed...

**Review by pstachel — APPROVED** [review]

**Review by CedrikNikita — APPROVED** [review]

File: src/popup/router/components/TransferReview.vue [hunk 3]
```diff
@@ -141,102 +141,97 @@ export default {
...
+    },
+    async transfer({ amount, recipient, selectedAsset }) {
+      this.loading = true;
+      try {
+        let actionResult;
 
-        if (!amountRaw || !recipient || !selectedAsset) {
-          return;
+        if (this.transferData.invoiceId != null) {
+          actionResult = await this.$store.dispatch('fungibleTokens/burnTriggerPoS', [
```
Comment (konradodo) [inline src/popup/router/components/TransferReview.vue:180]: mapAction

File: src/popup/router/components/TransferReview.vue [hunk 4]
```diff
@@ -141,102 +141,97 @@ export default {
...
+            selectedAsset.contractId,
+            amount,
+            this.transferData.invoiceContract,
+            this.transferData.invoiceId,
+            { waitMined: false, modal: false },
+          ]);
+        } else if (selectedAsset.contractId !== AETERNITY_CONTRACT_ID) {
+          actionResult = await this.$store.dispatch('fungibleTokens/transfer', [
```
Comment (konradodo) [inline src/popup/router/components/TransferReview.vue:188]: mapAction

File: src/popup/router/components/TransferReview.vue [hunk 5]
```diff
@@ -141,102 +141,97 @@ export default {
...
         }
 
-        const amount = (selectedAsset.contractId === AETERNITY_CONTRACT_ID)
-          ? aeToAettos(amountRaw)
-          : convertToken(amountRaw, selectedAsset.decimals);
-
-        if (this.isAddressUrl) {
-          this.sendTip({
+        if (actionResult && selectedAsset.contractId !== AETERNITY_CONTRACT_ID) {
+          this.$store.dispatch('addPendingTransaction', {
```
Comment (konradodo) [inline src/popup/router/components/TransferReview.vue:202]: mapAction

File: src/popup/router/components/TransferReview.vue [hunk 6]
```diff
@@ -141,102 +141,97 @@ export default {
...
+              callerId: this.account.address,
+              contractId: selectedAsset.contractId,
+              type: SCHEMA.TX_TYPE.contractCall,
+              function: 'transfer',
+            },
           });
-        } else {
-          this.transfer({
+        } else if (actionResult) {
+          this.$store.dispatch('addPendingTransaction', {
```
Comment (konradodo) [inline src/popup/router/components/TransferReview.vue:216]: mapAction

File: src/popup/router/components/TransferReview.vue [hunk 7]
```diff
@@ -277,20 +272,22 @@ export default {
...
         e.payload = { url: recipient };
         throw e;
       } finally {
         this.loading = false;
       }
-
-      this.$emit('success');
+    },
+    openTransactionFailedModal() {
+      this.$store.dispatch('modals/open', {
```
Comment (konradodo) [inline src/popup/router/components/TransferReview.vue:286]: mapAction

**Pull request merged**